### PR TITLE
Add Cygwin compatibility.

### PR DESCRIPTION
**Pull request by FGasper**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -350,3 +350,39 @@ jobs:
...
+    steps:
+      - name: Set up Cygwin
+        uses: cygwin/cygwin-install-action@master
+        with:
+            packages: make cmake gcc-g++ bash
+
+      - uses: actions/checkout@v3
+
+      - name: build
+        shell: C:\cygwin\bin\bash.exe --login --norc -eo pipefail -o igncr '{0}'
```
Comment (saghul) [inline .github/workflows/ci.yml]: You can put the shell in a default block to avoid having to repeat it.

**Review by saghul — APPROVED** [review]
LGTM with a suggestion.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (FGasper) [reply .github/workflows/ci.yml]: The shell is installed as part of the workflow, so I didn't think that would work. I'll try it, though.

Comment (FGasper) [reply .github/workflows/ci.yml]: It seems to work. Switched.

**Pull request merged**